### PR TITLE
Depth buffer fixes

### PR DESCRIPTION
**Pull request by GPUCode**
* Implements stencil testing and logic op
* Respects the depth write register
* Initial implementation of buffer clears

Improves 3D Land rendering
εικόνα

### CODE REVIEW

**Comment by wheremyfoodat** [comment]
So good

**Comment by ArbestRiAgain** [comment]
Yeah, previously it was just a gray screen.

File: src/core/renderer_gl/renderer_gl.cpp [hunk 1]
```diff
@@ -407,8 +483,39 @@ void RendererGL::clearBuffer(u32 startAddress, u32 endAddress, u32 value, u32 co
...
+			depthVal = (value & 0xffffff) / 16777215.0f;
+		}
+		depth->get().fbo.bind(OpenGL::DrawFramebuffer);
+		OpenGL::setDepthMask(true);
+		OpenGL::setClearDepth(depthVal);
+		OpenGL::clearDepth();
+		if (format == DepthFmt::Depth24Stencil8) {
+			const u8 stencil = (value >> 24);
+			OpenGL::setStencilMask(0xFF);
+			OpenGL::setClearStencil(stencil);
```
Comment (wheremyfoodat) [inline src/core/renderer_gl/renderer_gl.cpp:514]: Did somebody forget to actually clear stencil here :rboop:

**Comment by wheremyfoodat** [comment]
image

**Pull request merged**